server/src/views/frame.js

```diff
@@ -115,14 +64,14 @@ class Snippet extends React.Component {
     }
 
     return <div className="snippet-container">
-      <a href={'#clip=' + encodeURIComponent(clip.id)}>
+      <a href={'?clip=' + encodeURIComponent(clip.id)}>
         { node }
       </a>
       <p>
         <img ref="commentBubble" className="comment-bubble"
           src={ closed ? this.props.staticLink("img/comment-bubble.png") : this.props.staticLink("img/comment-bubble-open.png") }
           onClick={ this.onClickComment.bind(this) } />
-        <a href={'#clip=' + encodeURIComponent(clip.id)}>
+        <a href={'?clip=' + encodeURIComponent(clip.id)}>
```

Comment:
Because it's not a new resource; the clip that you are focused on is a classic use of a fragment.  We're doing something slightly fancier than scrolling, but conceptually it's the same.